### PR TITLE
Feature/child turns four email

### PR DESCRIPTION
**Pull request by dwybourn**
- Updated child turns four notification email to include correct payment amount.
- Refactored building of child turns four email personalisation out of message factory.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java [hunk 1]
```diff
@@ -46,6 +70,76 @@ private void sendChildTurnsFourEmail(PaymentCycle paymentCycle, NextPaymentCycle
...
+        Map<String, Object> emailPersonalisation = createPaymentEmailPersonalisationMap(paymentCycle, entitlementNextMonth);
+        emailPersonalisation.put(MULTIPLE_CHILDREN.getTemplateKeyName(), multipleChildrenTurningFourInMonth);
+        return EmailMessagePayload.builder()
+                .claimId(paymentCycle.getClaim().getId())
+                .emailType(EmailType.CHILD_TURNS_FOUR)
+                .emailPersonalisation(emailPersonalisation)
+                .build();
+    }
+
+    private Map<String, Object> createPaymentEmailPersonalisationMap(PaymentCycle paymentCycle, PaymentCycleVoucherEntitlement voucherEntitlement) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: suggest we name this something like `createEmailPersonalisationMapForNextCycle`

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java [hunk 2]
```diff
@@ -46,6 +70,76 @@ private void sendChildTurnsFourEmail(PaymentCycle paymentCycle, NextPaymentCycle
...
+        emailPersonalisation.put(NEXT_PAYMENT_DATE.getTemplateKeyName(), formattedCycleEndDate);
+        return emailPersonalisation;
+    }
+
+    private String getRegularPaymentAmount(PaymentCycleVoucherEntitlement voucherEntitlement) {
+        int totalVouchersForRegularPayment = getRegularPaymentVouchers(voucherEntitlement);
+        return formatRegularPaymentAmount(totalVouchersForRegularPayment, voucherEntitlement.getSingleVoucherValueInPence());
+    }
+
+    private String buildUnder4PaymentSummary(PaymentCycleVoucherEntitlement voucherEntitlement) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: Might want to add `ForNextCycle` to the end of all these paymentSummary methods.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandlerTest.java [hunk 3]
```diff
@@ -16,58 +16,99 @@
...
+
     @Test
-    void shouldSendEmailForChildTurnsFour() {
-        PaymentCycle paymentCycle = aValidPaymentCycle();
+    void shouldSendEmailForChildTurnsFourInFirstWeekOfNextCycle() {
         NextPaymentCycleSummary nextPaymentCycleSummary = NextPaymentCycleSummary.builder().numberOfChildrenTurningFour(1).build();
         given(childDateOfBirthCalculator.getNextPaymentCycleSummary(any())).willReturn(nextPaymentCycleSummary);
         //This entitlement specifically has no vouchers for 1-4 year olds.
-        PaymentCycleVoucherEntitlement nextEntitlement = aPaymentCycleVoucherEntitlementWithVouchersForUnderOne(START_OF_NEXT_CYCLE);
+        List<LocalDate> childrensDob = List.of(UNDER_ONE_ALL_OF_NEXT_CYCLE, TURNS_FOUR_DAY_ON_NEXT_PAYMENT_CYCLE);
```
Comment (YetAnotherMatt) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandlerTest.java]: `TURNS_FOUR_ON_DAY_ONE_OF_NEXT_PAYMENT_CYCLE`?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: Added ForNextCycle to the method name.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: Renamed.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandlerTest.java (hunk 3 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandlerTest.java]: Renamed.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java [hunk 4]
```diff
@@ -46,6 +70,76 @@ private void sendChildTurnsFourEmail(PaymentCycle paymentCycle, NextPaymentCycle
...
+        emailPersonalisation.put(PREGNANCY_PAYMENT.getTemplateKeyName(), buildPregnancyPaymentAmountSummary(voucherEntitlement));
+        emailPersonalisation.put(CHILDREN_UNDER_1_PAYMENT.getTemplateKeyName(), buildUnder1PaymentSummary(voucherEntitlement));
+        emailPersonalisation.put(CHILDREN_UNDER_4_PAYMENT.getTemplateKeyName(), buildUnder4PaymentSummaryForNextCycle(voucherEntitlement));
+        emailPersonalisation.put(REGULAR_PAYMENT.getTemplateKeyName(), getRegularPaymentAmount(voucherEntitlement));
+        String formattedCycleEndDate = paymentCycle.getCycleEndDate().plusDays(1).format(DATE_FORMATTER);
+        emailPersonalisation.put(NEXT_PAYMENT_DATE.getTemplateKeyName(), formattedCycleEndDate);
+        return emailPersonalisation;
+    }
+
+    private String getRegularPaymentAmount(PaymentCycleVoucherEntitlement voucherEntitlement) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: `getRegularPaymentAmountForNextCycle`

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java [hunk 5]
```diff
@@ -46,6 +70,76 @@ private void sendChildTurnsFourEmail(PaymentCycle paymentCycle, NextPaymentCycle
...
+
+    private int getVouchersForBetweenOneAndFourForFinalWeek(PaymentCycleVoucherEntitlement voucherEntitlement) {
+        return voucherEntitlement.getLastVoucherEntitlementForCycle().getVouchersForChildrenBetweenOneAndFour();
+    }
+
+    private int getRegularPaymentVouchers(PaymentCycleVoucherEntitlement voucherEntitlement) {
+        return voucherEntitlement.getLastVoucherEntitlementForCycle().getTotalVoucherEntitlement() * numberOfCalculationPeriods;
+    }
+
+    private String buildPregnancyPaymentAmountSummary(PaymentCycleVoucherEntitlement voucherEntitlement) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: `buildPregnancyPaymentAmountSummaryForNextCycle`

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java [hunk 6]
```diff
@@ -46,6 +70,76 @@ private void sendChildTurnsFourEmail(PaymentCycle paymentCycle, NextPaymentCycle
...
+        return formatPaymentAmountSummary(
+                "\n* %s for a pregnancy",
+                voucherEntitlement.getVouchersForPregnancy(),
+                voucherEntitlement.getSingleVoucherValueInPence());
+    }
+
+    private String buildUnder1PaymentSummary(PaymentCycleVoucherEntitlement voucherEntitlement) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: `buildUnder1PaymentSummaryForNextCycle`

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java (hunk 4 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: A method called getRegularPaymentAmountForNextCycle already exists, I've decided to inline this method as it's only one line.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java (hunk 5 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: Renamed.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java (hunk 6 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/PaymentCycleEmailHandler.java]: Renamed.

**Pull request merged**